src/script/ui/state/rdkitjs/index.js

```diff
@@ -90,15 +91,44 @@ function deleteAllSGroupsWithName(restruct, action, fieldName) {
 	});
 }
 
-function doTransformation(method, mol) {
+/**
+ * @param method {string}
+ * @param mol {*} TODO (pradeep): Figure out a typedef for mol
+ * @param molDetails {StructDetails}
+ * @return {*}
+ */
+function doTransformation(method, mol, molDetails) {
 	switch (method) {
 		case 'aromatize':
 			return mol.get_aromatic_form();
 		case 'dearomatize':
 			return mol.get_kekule_form();
 		case 'clean':
 			return mol.get_new_coords();
+		case 'toggleExplicitHydrogen':
+			if (molDetails.hasExplicitHydrogen) {
+				return mol.remove_hs();
+			} else {
+				return mol.add_hs();
+			}
 		default:
 			return '';
 	}
 }
+
+/**
+ * @param struct {Struct}
+ * @return {StructDetails}
+ */
+function getStructDetails(struct) {
+	return {
+		hasExplicitHydrogen: !!struct.hasExplicitHydrogen(),
```

Comment:
post this comment - https://github.com/schrodinger/ketcher/pull/16/files#r394796405, we could get rid of the double negation. Also, does it then seem worth extracting into a separate 'getStructDetails'? Maybe we could simply inline it there.

Reply:
removed, we'll make it into a function if it's actually required at that point